superset-frontend/packages/superset-ui-chart-controls/src/operators/sortOperator.ts

```diff
@@ -28,42 +28,40 @@ import {
 import { PostProcessingFactory } from './types';
 import { extractExtraMetrics } from './utils';
 
-export const sortOperator: PostProcessingFactory<PostProcessingSort> = (
-  formData,
-  queryObject,
-) => {
-  // the sortOperator only used in the barchart v2
-  const sortableLabels = [
-    getXAxisLabel(formData),
-    ...ensureIsArray(formData.metrics).map(getMetricLabel),
-    ...extractExtraMetrics(formData).map(getMetricLabel),
-  ].filter(Boolean);
+export const sortOperator: PostProcessingFactory<PostProcessingSort> =
+  formData => {
+    // the sortOperator only used in the barchart v2
+    const sortableLabels = [
+      getXAxisLabel(formData),
+      ...ensureIsArray(formData.metrics).map(getMetricLabel),
+      ...extractExtraMetrics(formData).map(getMetricLabel),
+    ].filter(Boolean);
+
+    if (
+      hasGenericChartAxes &&
+      isDefined(formData?.x_axis_sort) &&
+      isDefined(formData?.x_axis_sort_asc) &&
+      sortableLabels.includes(formData.x_axis_sort) &&
+      // the sort operator doesn't support sort-by multiple series.
+      isEmpty(formData.groupby)
+    ) {
+      if (formData.x_axis_sort === getXAxisLabel(formData)) {
+        return {
+          operation: 'sort',
+          options: {
+            is_sort_index: true,
+            ascending: formData.x_axis_sort_asc,
+          },
+        };
+      }
 
-  if (
-    hasGenericChartAxes &&
-    isDefined(formData?.x_axis_sort) &&
-    isDefined(formData?.x_axis_sort_asc) &&
-    sortableLabels.includes(formData.x_axis_sort) &&
-    // the sort operator doesn't support sort-by multiple series.
-    isEmpty(formData.groupby)
-  ) {
-    if (formData.x_axis_sort === getXAxisLabel(formData)) {
       return {
         operation: 'sort',
         options: {
-          is_sort_index: true,
+          by: formData.x_axis_sort,
```

Comment:
Ummm.... I'm not sure how this got changed... it's certainly not a linting rule.